3rdparty/carotene/hal/CMakeLists.txt

```diff
@@ -55,7 +55,7 @@ endif()
 set(CAROTENE_NS "carotene_o4t" CACHE STRING "" FORCE)
 
 function(compile_carotene)
-  if(";${CPU_BASELINE_FINAL};" MATCHES ";NEON;")
+  if(";${CPU_BASELINE_FINAL};" MATCHES ";NEON;" AND CV_ENABLE_INTRINSICS)
```

Comment:
`CV_ENABLE_INTRINSICS` is primary about OpenCV code (to help with debugging).

IMHO, we should not spread OpenCV-specific flags on 3rdparty code.

---

`CPU_BASELINE_FINAL` is about compiler capabilities.

Reply:
IMHO, Carotene should be trieted as 3rd party library and CV_ENABLE_INTRINSICS should not affect it. It's main goal is to bring HAL with NEON. Without NEON it's mostly useless. Please build OpenCV without Carotene, please use `-DWITH_CAROTENE=OFF` in CMake.

Reply:
Could we have a single global flag that guarantee all optimization are turned off? Since CAROTENE is turned on automatically when building for Android, to avoid confusion how about making CV_DISABLE_OPTIMIZATION turning CAROTENE off as well?

Reply:
Here for `WITH_CAROTENE` flag: https://github.com/opencv/opencv/blob/4.5.5/CMakeLists.txt#L246

`ON` => `(NOT CV_DISABLE_OPTIMIZATION)`

(similar to `WITH_EIGEN` option few lines below)

---

Ensure to clean CMake cache between experiments.

---

This patch should go into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first.
We will merge changes from 3.4 into 4.x regularly (weekly/bi-weekly).

Please:
- change "base" branch of this PR: 4.x => 3.4 (use "Edit" button near PR title)
- rebase your commits from 4.x onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/4.x`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".